src/language/mod.rs

```diff
@@ -106,3 +117,20 @@ impl AddAssign for Language {
     }
 }
 
+impl<'a> Add<&'a Self> for Language {
+    type Output = Self;
+
+    fn add(self, rhs: &Self) -> Self {
+        let mut stats = self.stats;
+        stats.extend(rhs.stats.clone());
```

Comment:
The `clone` here is unfortunate, the only way I see around it is for stats to be `Vec<Rc<Stat>>`, do you have a better solution?